src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -102,9 +112,12 @@ def shrinkLargeSummary(couchDbInstance, sizeLimit):
                 msg += f"has {fieldBytes} bytes under the attr: {innerAttr}. Truncating it.."
                 logging.warning(msg)
                 if innerAttr == "errors":
-                    # keep only the first 5k characters
-                    doc[innerAttr] = doc[innerAttr][:5000]
-                    doc[innerAttr] += "... error message truncated ..."
+                    # keep only the first 5k characters of the error detail
```

Comment:
At the moment the initial error dictionary has been created, was it assured, the next two fields used in the following two `for` cycles are actually iterrable (even though with zero length)?

Reply:
Yes, data type defaults to list.